### PR TITLE
Import `bundle gem` generated scaffolding

### PR DESCRIPTION
**Pull request by notCalle**
Import the scaffolding that would have been created if we initialized the project with `bundle gem`, instead of starting from scratch. This adds project support for testing and release management, including publishing to a gem server, with the help of `rake`.

Restructure the palletjack-tools files to match bundler requirements and conventions.

### CODE REVIEW

File: palletjack.gemspec [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+  spec.require_paths = ["lib"]
+
+  spec.platform	= Gem::Platform::RUBY
+  spec.required_ruby_version = '~>2'
+
+  spec.add_runtime_dependency 'activesupport', '~>4'
+  spec.add_runtime_dependency 'rugged', '~> 0.24'
+  spec.add_runtime_dependency 'kvdag', '~>0.1'
+
+  spec.add_development_dependency "bundler", "~> 1.13"
```
Comment (creideiki) [inline palletjack.gemspec]: Is 1.13 really necessary? CentOS 7 gives us 1.7.8, so it would be advantageous if that worked.

File: README.md [hunk 2]
```diff
@@ -190,6 +206,19 @@ system:
...
+After checking out the repo, run `bin/setup` to install
+dependencies. Then, run `rake spec` to run the tests. You can also run
+`bin/console` for an interactive prompt that will allow you to
+experiment.
+
+To install this gem onto your local machine, run `bundle exec rake
+install`. To release a new version, update the version number in
+`version.rb`, and then run `bundle exec rake release`, which will
+create a git tag for the version, push git commits and tags, and push
+the `.gem` file to [rubygems.org](https://rubygems.org).
```
Comment (creideiki) [inline README.md]: Is there a point to pasting generic release instructions which will not work with our processes into this specific file?

File: palletjack.gemspec [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
+  spec.add_development_dependency "rake", "~> 10.0"
```
Comment (creideiki) [inline palletjack.gemspec]: Same here. CentOS 7 has 0.9.6.

File: palletjack.gemspec [hunk 4]
```diff
@@ -1,22 +1,36 @@
...
+  spec.add_development_dependency "rspec", "~> 3.0"
```
Comment (creideiki) [inline palletjack.gemspec]: Same here. CentOS 7 has 2.14.1.

**Review by creideiki — CHANGES_REQUESTED** [review]

File: palletjack.gemspec (hunk 1 above)
Comment (notCalle) [reply palletjack.gemspec]: Changed to ~>1.7, and it seems to still work.

File: palletjack.gemspec (hunk 3 above)
Comment (notCalle) [reply palletjack.gemspec]: Changed to ~>0.9, and it seems to still work.

File: palletjack.gemspec (hunk 4 above)
Comment (notCalle) [reply palletjack.gemspec]: Changed to ~>2.14, and it seems to still work.

File: README.md (hunk 2 above)
Comment (notCalle) [reply README.md]: Not really, just chunk merged the boilerplate stuff with what was there.

**Comment by creideiki** [comment]
Dependency versions should be downgraded in `tools/palletjack-tools.gemspec` as well.

**Review by creideiki — APPROVED** [review]

**Pull request merged**